lua/treesitter-context.lua

```diff
@@ -295,38 +295,21 @@ function M.update_context()
   end
 end
 
-local async_wrap = function(f)
-  local wrapped = function(...)
-    local handle
-    handle = vim.loop.new_async(vim.schedule_wrap(function(...)
-      f(...)
-      handle:close()
-    end))
-    handle:send(...)
-  end
-
-  return wrapped
-end
-
```

Comment:
We shouldn't need any of this because `vim.defer_fn` automatically wraps the callback. See https://neovim.io/doc/user/lua.html#vim.defer_fn